src/libcsg/modules/io/xmltopologyreader.cc

```diff
@@ -224,7 +224,6 @@ void XMLTopologyReader::ParseMolecule(Property &p, string molname, int nbeads,
     }
     resnr++;
   }
-  _mol_index++;
```

Comment:
If it is meant to be the residue number shouldn't this line be changed:

   bname << _mol_index << ":" << molname << ":" << b.name;

to:

   bname << resnr << ":" << molname << ":" << b.name;

Reply:
@JoshuaSBrown is right, it should be the string should `<local residue nr>:molname:`<atom name>. Maybe in xml reader `mol_index` and `resnr` is the same?

However in #369, `2:Pt:C451` was wrong as `C451` was part of the 1st residue in `Pt`.

Reply:
How the residues are counted for Lammps and Gromacs? I have never had to use in mapping 1:NAME, 2:NAME;

Reply:
I am not sure how residues are handled by Gromacs or LAMMPS, the only place I have seen them used a lot is in pdb files. The use of residues is VOTCA is a bit misleading as internally residues are also treated as coarse grained components of a given molecule, but this is not consistent with the actual description of a residue in the literature. At times the use of a residue number is also internally treated as the same thing as a molecule number, hence the confusion. @junghans is my understanding correct?

Reply:
We could make it consistent, if you have like. We just need to ensure multiple residues per molecule are supported.

Reply:
Yes, I would like to make it consistent. We need to clarify the description of residue, coarse grained bead and molecule and stick to it. I will create an issue, but this is not a high priority.

Reply:
https://github.com/votca/csg/issues/375